internal/controllers/common_controller.go

```diff
@@ -117,3 +117,22 @@ func setDeprecationStatusesUnknown(conditions *[]metav1.Condition, message strin
 		})
 	}
 }
+
+// setInstalledAndHealthyFalse sets the Installed and if the feature gate is enabled, the Healthy conditions to False,
+// and allows to set the Installed condition reason and message.
+func setInstalledAndHealthyFalse(conditions *[]metav1.Condition, message string, generation int64) {
```

Comment:
Nit: Had added status updates in `cluster extension_status.go`. But shouldn't matter.

Reply:
There were also a bunch in this file...